lib/Dialect/LLHD/Transforms/LowerProcesses.cpp

```diff
@@ -103,8 +103,6 @@ bool Lowering::matchControlFlow() {
   auto skipToMergePoint = [&](Block *block) -> std::pair<Block *, ValueRange> {
     ValueRange operands;
     while (auto branchOp = dyn_cast<cf::BranchOp>(block->getTerminator())) {
-      if (!block->without_terminator().empty())
-        break;
```

Comment:
That would allow arbitrary ops in the entry block, right? What if there's some DPI call in there? We would have to put that into a separate initial op and funnel the (potential) result through to the lowered combinational op to make sure it is only called once at simulation start unless the sequence of blocks in the infinite loop has the same call that is skipped in the initial startup?
So, I think it would be fine to relax this constraint to checking that ALL ops in the blocks before the converged block are equivalent, rather than just the ones feeding the block arguments.

Reply:
Hmmm yeah good point. Let me maybe limit this to side-effect free ops for now, since the op-by-op comparison is a bit annoying in the general case when there are a different number of blocks between the entry and the convergence point, and the wait op an d the convergence point.